### PR TITLE
Pin Terraform version - [CORE-1297]

### PR DESCRIPTION
**Pull request by arsci**
## Description

* Pin Terraform version `<= 1.5.7`
* Update Readme
* Update `instance-type` example to use newer ubuntu image (`16.04` is no longer available).

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [x] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes

Added Terraform version upper restriction

### CODE REVIEW

**Comment by arsci** [comment]
`upgrade-test` is expected to fail here. The previous image use to test the `instance-type` filter no longer exists so `main` is broken until the new filter in this PR is merged.

File: README.md [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
+![Terraform Version](https://img.shields.io/badge/tf-%3E%3D1.1.0%3C%2C%20%3C%3D1.5.7-blue.svg)
 
 # Terraform Utility Modules
 
 This repo contains miscellaneous utility and helper modules for use with Terraform.
 
+## Information on HashiCorp BSL License Change
+
+Due to the HashiCorp BSL license change, we are restricting Terraform to the latest open source version (`1.5.7`). 
+Going forward we will be adding support for OpenTofu. For additional information on OpenTofu please see the
```
Comment (MoonMoon1919) [inline README.md:12]: In the docs I went with the following call to action for opentofu (after stating we'd support it) -

```markdown
To learn more, see the official [OpenTofu website](https://opentofu.org/) and [project status](https://github.com/opentofu/opentofu/blob/main/WEEKLY_UPDATES.md).
```

**Review by MoonMoon1919 — COMMENTED** [review]
Added a comment on the readme notice, but otherwise LGTM

**Review by MoonMoon1919 — APPROVED** [review]
lgtm!

**Pull request merged**